From 7b2fd8b3f59b8ccce3b8df3f11e528e934866b41 Mon Sep 17 00:00:00 2001
From: Binyuan Lan <lby@rock-chips.com>
Date: Thu, 29 Sep 2016 20:01:55 +0800
Subject: [PATCH] usb: phy-rockchip-inno-usb2: fix wrong charging state when
 otg host connect

No need notify charging-external-connector state when otg host connect.

Change-Id: I1d5c6e4fb2ad504f169ef0fd5b82b06f31783922
Signed-off-by: Binyuan Lan <lby@rock-chips.com>
---
 drivers/phy/phy-rockchip-inno-usb2.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/phy/phy-rockchip-inno-usb2.c b/drivers/phy/phy-rockchip-inno-usb2.c
index 6b819162ac22..644c61d15db2 100644
--- a/drivers/phy/phy-rockchip-inno-usb2.c
+++ b/drivers/phy/phy-rockchip-inno-usb2.c
@@ -678,9 +678,9 @@ static void rockchip_usb2phy_otg_sm_work(struct work_struct *work)
 			rport->state = OTG_STATE_B_IDLE;
 			rockchip_usb2phy_power_off(rport->phy);
 		}
-		break;
+		return;
 	default:
-		break;
+		return;
 	}
 
 	if (extcon_get_state(rphy->edev, cable) != rport->vbus_attached)
-- 
2.35.3

